code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -44,10 +48,44 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
+data ComponentIdentifier = Hackage Text | GHC GHCComponent
+  deriving stock (Show, Eq)
+
+-- Keep this list in sync with the 'ghcComponentFromText' below
+data GHCComponent = GHCCompiler | GHCi | GHCRTS | GHCPkg | RunGHC | IServ | HP2PS | HPC | HSC2HS | Haddock
+  deriving stock (Show, Eq, Enum, Bounded)
+
+ghcComponentToText :: GHCComponent -> Text
+ghcComponentToText c = case c of
+  GHCCompiler -> "compiler"
+  GHCi -> "ghci"
+  GHCRTS -> "rts"
+  GHCPkg -> "ghc-pkg"
+  RunGHC -> "runghc"
+  IServ -> "iserv"
+  HP2PS -> "hp2ps"
+  HPC -> "hpc"
+  HSC2HS -> "hsc2hs"
+  Haddock -> "haddock"
+
+ghcComponentFromText :: Text -> Maybe GHCComponent
+ghcComponentFromText c = case c of
+  "compiler" -> Just GHCCompiler
```

Comment:
I wonder if we should just use the string `"ghc"` instead of `"compiler"`, so that in all relevant cases the string is just the executable name.  This would also mean using `"ghc-iserv"` instead of `"iserv"`.

The only case that does not correspond to a particular executable is then "rts".  This is fine (we do need to represent it).

Reply:
Good idea, this is now implemented.